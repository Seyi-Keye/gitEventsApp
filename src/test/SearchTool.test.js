import React from "react";
import SearchTool from "../SearchTool";
import renderer from "react-test-renderer";

describe("SearchTool component", () => {
  let wrapper = renderer.create(<SearchTool />);
  it("renders succesfully", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("renders without props", () => {
    const testInstace = wrapper.root;
    expect(testInstace.props).toEqual({});
  });

  it("renders correct initial state", () => {
    const testInstance = wrapper.root;
    expect(testInstance.instance.state).toEqual({ username: "" });
  });
});

describe("SearchTool component", () => {
  let wrapper = renderer.create(<SearchTool />);
  const testInstance = wrapper.root.instance;
  testInstance.setState({ username: "aromokeye" });

  it("renders succesfully with props", () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("renders with props", () => {
    console.log(testInstance);
    expect(testInstance.state).toEqual({ username: "aromokeye" });
  });
});

fdescribe("SearchTool functionalities", () => {
  // let handleFormSubmit = false;
  // const node = {
  //   createNodeMock: element => {
  //     if (element.type === "button") {
  //       // mock a focus function
  //       return {
  //         handleFormSubmit: () => {
  //           focused = true;
  //         }
  //       };
  //     }
  //     return null;
  //   }
  // }

  const stub = () => true;

  const componentRoot = renderer.create(<SearchTool onClick={stub} />).root;

  const componentInstance = componentRoot.instance;
  componentInstance.setState({ username: "aromokeye" });
  const button = componentRoot.findByType("button").props;
  console.log(
    "button",
    componentRoot.findByType("button").props.onClick(),
    componentInstance
  );
  it("calls handleFormSubmit on button click", () => {
    expect(
      componentRoot.findByType(button).props.handleFormSubmit()
    ).toHaveBeenCalledTimes(1);
  });
});
