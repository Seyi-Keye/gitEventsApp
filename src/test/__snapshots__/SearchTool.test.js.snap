exports[`SearchTool component renders succesfully 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "margin": "auto",
      "width": "100%",
    }
  }>
  <form
    onSubmit={[Function]}>
    <input
      name="githubUsername"
      onChange={[Function]}
      placeholder="Type github username"
      style={
        Object {
          "height": "30px",
          "width": "20%",
        }
      }
      type="text"
      value="" />
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "blue",
          "color": "white",
          "height": "30px",
          "width": "10%",
        }
      }
      type="button">
       
      Search Github Events
    </button>
  </form>
</div>
`;

exports[`SearchTool component renders succesfully with props 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "margin": "auto",
      "width": "100%",
    }
  }>
  <form
    onSubmit={[Function]}>
    <input
      name="githubUsername"
      onChange={[Function]}
      placeholder="Type github username"
      style={
        Object {
          "height": "30px",
          "width": "20%",
        }
      }
      type="text"
      value="aromokeye" />
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "blue",
          "color": "white",
          "height": "30px",
          "width": "10%",
        }
      }
      type="button">
       
      Search Github Events
    </button>
  </form>
</div>
`;
